### PR TITLE
Global 'java.jdt.ls.vmargs' setting is ignored

### LINKED ISSUE #1175: Version 0.54 Not Using Global "java.jdt.ls.vmargs" Setting
As of the latest `0.54` update, the `java.jdt.ls.vmargs` setting, if set globally (i.e., user settings), doesn't seem to be used. I noticed this in my project because I use Project Lombok, and the global vmargs are set accordingly:

```
"java.jdt.ls.vmargs": "-noverify -Xmx1G -XX:+UseG1GC -XX:+UseStringDeduplication -javaagent:\"/home/vagrant/.vscode-server/extensions/gabrielbb.vscode-lombok-0.9.9/server/lombok.jar\"",
```

If I copy this exact setting and place it in my workspace/project folder (i.e., in `.vscode/settings.json`), Lombok annotations work as expected.

I've reverted back to `0.53.1`, and it works as expected, utilizing the global `java.jdt.ls.vmargs` setting.

##### Environment
- Operating System: Windows 7 w/ VS Code Remote, Connecting to Ubuntu 18.04
- JDK version: `openjdk version "1.8.0_222"`
- Visual Studio Code version: 
- Java extension version: `1.40.2`

##### Steps To Reproduce
1. Verify version `0.54` of the `vscode-java` extension is installed.
1. Follow the instructions on [vscode-java > Lombok Support](https://github.com/redhat-developer/vscode-java/wiki/Lombok-support), which may be just installing the Lombok extension for VS Code. Note this will require a reload/restart of VS Code.
1. Open a Java project that utilizes [Project Lombok[(https://projectlombok.org/).
1. Open any class that uses Lombok annotations, errors will appear relating to them (e.g., can't find the `log` variable if using loggers)

You can then revert the `vscode-java` extension to `0.53.1`, reload/restart VS Code, and the Lombok-related errors will disappear.

##### Logs
When viewing the server log, it seems to be picking up the setting when it's global...so not sure what is happening behind the scenes that it's not actually applying the setting?

```
!ENTRY org.eclipse.jdt.ls.core 1 0 2019-12-04 14:56:42.050
!MESSAGE >>New configuration: {java={home=/usr/lib/jvm/java-8-openjdk-amd64, jdt={ls={vmargs=-noverify -Xmx1G -XX:+UseG1GC -XX:+UseStringDeduplication -javaagent:"/home/vagrant/.vscode-server/extensions/gabrielbb.vscode-lombok-0.9.9/server/lombok.jar"}}, errors={incompleteClasspath={severity=warning}}, configuration={checkProjectSettingsExclusions=false, updateBuildConfiguration=interactive, maven={userSettings=null}}, trace={server=messages}, import={maven={enabled=true}, gradle={enabled=true, wrapper={enabled=true}, version=null, arguments=null, jvmArguments=null, home=null}, exclusions=[**/node_modules/**, **/.metadata/**, **/archetype-resources/**, **/META-INF/maven/**]}, maven={downloadSources=false, updateSnapshots=false}, referencesCodeLens={enabled=false}, signatureHelp={enabled=false}, implementationsCodeLens={enabled=false}, format={enabled=true, settings={url=null, profile=null}, comments={enabled=true}, onType={enabled=true}}, saveActions={organizeImports=false}, contentProvider={preferred=null}, autobuild={enabled=true}, maxConcurrentBuilds=1.0, completion={enabled=true, overwrite=true, guessMethodArguments=false, favoriteStaticMembers=[org.junit.Assert.*, org.junit.Assume.*, org.junit.jupiter.api.Assertions.*, org.junit.jupiter.api.Assumptions.*, org.junit.jupiter.api.DynamicContainer.*, org.junit.jupiter.api.DynamicTest.*, org.mockito.Mockito.*, org.mockito.ArgumentMatchers.*, org.mockito.Answers.*], filteredTypes=[java.awt.*, com.sun.*], importOrder=[java, javax, com, org]}, foldingRange={enabled=true}, progressReports={enabled=true}, codeGeneration={hashCodeEquals={useJava7Objects=false, useInstanceof=false}, useBlocks=false, generateComments=false, toString={template=${object.className} [${member.name()}=${member.value}, ${otherMembers}], codeStyle=STRING_CONCATENATION, skipNullValues=false, listArrayContents=true, limitElements=0.0}}, selectionRange={enabled=true}}}
```

##### Current Result
Lombok annotations don't work if configured globally per [vscode-java > Lombok Support](https://github.com/redhat-developer/vscode-java/wiki/Lombok-support).

##### Expected Result
Lombok annotations work if configured globally, as seen version `0.53.1` of this extension.

##### Additional Informations
None

### PR DESCRIPTION
**Pull request by snjeza**
Fixes #1175 

Signed-off-by: Snjezana Peco <snjezana.peco@redhat.com>

### CODE REVIEW

File: src/javaServerStarter.ts [hunk 1]
```diff
@@ -72,6 +72,8 @@ function prepareParams(requirements: RequirementsData, javaConfiguration, worksp
 				vmargsCheck = workspace.getConfiguration().inspect('java.jdt.ls.vmargs').globalValue;
 			}
 		}
+	} else {
+		vmargsCheck = workspace.getConfiguration().inspect('java.jdt.ls.vmargs').globalValue;
```
Comment (fbricon) [inline src/javaServerStarter.ts]: workspace.getConfiguration().get('java.jdt.ls.vmargs', '');

Comment (snjeza) [reply src/javaServerStarter.ts]: Fixed.

Comment (fbricon) [reply src/javaServerStarter.ts]: Not at all. This returns an Object.
getJavaConfiguration().get('jdt.ls.vmargs') works

Comment (snjeza) [reply src/javaServerStarter.ts]: I have updated the PR.

File: src/settings.ts [hunk 2]
```diff
@@ -166,7 +166,7 @@ export async function checkJavaPreferences(context: ExtensionContext) {
 	if (isVerified) {
 		return javaHome;
 	} else {
-		return workspace.getConfiguration().inspect<string>('java.home').globalValue;
+		return getJavaConfiguration().get('home');
```
Comment (fbricon) [inline src/settings.ts]: no, if java.home is not verified, or warning is ignored, you want to avoid picking the workspace value. Stick with the global one

Comment (snjeza) [reply src/settings.ts]: Fixed.

**Comment by fbricon** [comment]
can you please restore this line from the previous commit, it was valid: 
https://github.com/redhat-developer/vscode-java/commit/4bfc73dc201622ef70785f648d18db47756bf966#diff-f953b3a2fb6d7cd16d698364c6a1dc45R128

**Comment by snjeza** [comment]
@fbricon I have updated the PR.

**Comment by fbricon** [comment]
Thanks @snjeza!

**Pull request merged**